### PR TITLE
Improve the microwave

### PR DESCRIPTION
**Pull request by flappybatpal**
[REWORK]

## About the PR 

1. Adds a start sound to the microwave.  https://freesound.org/people/JarredGibb/sounds/219483/ used under [CC0] https://creativecommons.org/publicdomain/zero/1.0/)
2. Allows the sponge to be used to clean a dirty microwave.
3. Puts ingredients into contents and allows them to be emptied out if not used.
4. Dump all contents except valid ingredient types after operation for safety. The old microwave would steal anything cooked with a valid recipe.
5. Adds message if the microwave is run while empty.
6. Removes duplicate code and implements procs.
7. Use action bars for cleaning and repairing.
8. Add some text hints on how to clean/repair.
9. Lists the unusual item in the menu.
10. Block anything being added to the microwave while it is operating.

## Why's this needed? 
Microwave code needed improving.
This doesn't deal with the recipe and ingredient handling

```changelog
(u)Flappybat
(+)Microwaves now let you empty out the contents instead of throwing them away.
(+)Microwaves now can be cleaned with a sponge.
```

### CODE REVIEW

**Comment by flappybatpal** [comment]
To Do:

1. Action bars on repairing/cleaning
2. Give users advice on repairing/cleaning
3. Check dice functionality
4. Double check behaviour compared to master

File: code/obj/machinery/microwave.dm [hunk 1]
```diff
@@ -17,7 +23,9 @@
 	var/operating = 0 // Is it on?
 	var/dirty = 0 // Does it need cleaning?
 	var/broken = 0 // How broken is it???
+	var/cook_time = 200 // The time to wait before spawning the item
```
Comment (ZeWaka) [inline code/obj/machinery/microwave.dm:26]: 200 what

Comment (ZeWaka) [reply code/obj/machinery/microwave.dm:26]: code doc format also

**Review by virvatuli-gh — APPROVED** [review]
sound good and it Works On My Machine

File: code/obj/machinery/microwave.dm [hunk 2]
```diff
@@ -17,7 +23,9 @@
...
 	var/list/available_recipes = list() // List of the recipes you can use
+	var/datum/recipe/cooked_recipe = null // The recipe being cooked
```
Comment (ZeWaka) [inline code/obj/machinery/microwave.dm:28]: code doc format

**Review by ZeWaka — CHANGES_REQUESTED** [review]

File: code/obj/machinery/microwave.dm (hunk 1 above)
Comment (flappybatpal) [reply code/obj/machinery/microwave.dm:26]: Sorry not sure what code doc format means here. I don't see anything relevant in the Goonstation Code Guide?

File: code/obj/machinery/microwave.dm (hunk 2 above)
Comment (flappybatpal) [reply code/obj/machinery/microwave.dm:28]: As other comment, not sure what is wanted here.

File: code/obj/machinery/microwave.dm (hunk 1 above)
Comment (ZeWaka) [reply code/obj/machinery/microwave.dm]: https://spacestation13.github.io/DMByExample/meta/dmdoc.html

Comment (flappybatpal) [reply code/obj/machinery/microwave.dm]: Ok, I think this is done correctly. Have tried replacing appropriate parts.

File: code/obj/machinery/microwave.dm (hunk 2 above)
Comment (flappybatpal) [reply code/obj/machinery/microwave.dm]: Ok, I think this is done correctly. Have tried replacing appropriate parts.

**Review by ZeWaka — APPROVED** [review]

**Pull request merged**